doc/api/process.md

```diff
@@ -1917,6 +1917,44 @@ console.log('After:', getActiveResourcesInfo());
 //   After: [ 'TTYWrap', 'TTYWrap', 'TTYWrap', 'Timeout' ]
 ```
 
+## `process.getBuiltin(id)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `id` {string} ID of the built-in module being requested. Must not contain the `node:` prefix.
```

Comment:
I am not actually sure if forbidding `node:` is a good idea, but then it seems a bit weird to allow it since the whole point of the `node:` prefix is to prevent ambiguity in `require()` that may be used to load user modules. But with `process.getBuiltin()` which can only be used to load built-ins, such ambiguity does not exist...

Reply:
I don’t see the point on erroring for the `node:` prefix. Permitting it allows using the same set of builtin specifiers for this and for `import()`.

Reply:
The weird exception arises when you can `process.getBuiltin('test')` but not `require('test')` (or they don't return the same thing if you don't happen to have a module called `test`).

Reply:
Another way to make it less weird would be to only allow `node:` IDs, but then it somehow still feels weird - this is only for loading built-ins, why all the `node:` to disambiguate?

Reply:
Per https://nodejs.org/api/module.html#moduleisbuiltinmodulename:

```js
import { isBuiltin } from 'node:module';
isBuiltin('node:fs'); // true
```

If this works, then `process.getBuiltin('node:fs')` needs to work too.

Reply:
That goes back to the `node:test` exception, because `module.isBuiltin('test')` returns false. That means `process.getBuiltin('test')` should throw?

Reply:
And if `process.getBuiltin('test')` throws, that encourage people again to do `node:` always even with `process.getBuiltin()`, which then feels a bit strange because `process.getBuiltin()` does not load any user modules and there is nothing to disambiguate...

Reply:
That's fine. It's better to be consistent with the other builtin APIs.

Reply:
> It's better to be consistent with the other builtin APIs.

There is the more widely used `module.builtinModules` (also used in the tests) which doesn't include `node:` names...a quick search with `module.builtinModules.includes` that many popular projects are using it to determine whether an ID is a builtin https://github.com/search?q=module.builtinModules.includes&type=code

Reply:
> There is the more widely used `module.builtinModules` (also used in the tests) which doesn’t include `node:` names

That includes neither `test` nor `node:test`. But I don’t think we want to have `getBuiltin` refuse to retrieve the test runner, do we? (Side note, `module.builtinModules` probably _should_ include `node:test`, I’d think.)

I think the relevant equivalent is `module.isBuiltin`. Every input that returns `true` from that should be accessible from `process.getBuiltin`. I think most users would consider it a bug for `isBuiltin('node:test')` to return true and `getBuiltin('node:test')` to throw.

Reply:
I think the more important question is, should `process.getBuiltin('test')` throw?

Reply:
> I think the more important question is, should `process.getBuiltin('test')` (and any future prefix-ed builtins) throw?

Yes, because `require('node:module').isBuiltin('test')` returns `false`. As in, there is no builtin named `test`. There’s only a builtin named `node:test`. It’s that simple, and it’s not a bug.

If you want to _additionally_ support `'test'` here, then it should get supported by `isBuiltin` first. The two APIs should support the same set of strings (as should `module.builtinModules`, but that’s a separate PR).

Reply:
Just my two cents, given we decided to always use the `node:` prefix for future built-in modules, wouldn't it be preferable to always require it? I.e., `process.getBuiltin('fs')` would throw, but `process.getBuiltin('node:fs')` would work – and `process.getBuiltin('node:test') would also work.

Reply:
TBH I don't really know which to choose. I could go with no `node:`, always `node:`, or sometimes `node:`, but I am not super confident that they are the choices we won't regret about.....should we do a poll on this somehow?

Reply:
Why is this complicated? `isBuiltin` and `getBuiltin` should support exactly the same set of strings—no more, no less. Anything else would be inconsistent. And separately, we should fix the inconsistency of `builtinModules`. We don’t need a poll to know that our APIs should be consistent.

Reply:
It's entirely likely that would implement support for this in worker's node.js compat mode. In that case, I would actually prefer the `node:` be allowed since we generally prefer the specificity.

Reply:
Consistency +++1

That said, `getBuiltin('node:')` is basically the same as `getBuiltin('builtin:')`, which seems a bit silly.

Reply:
It’s obviously redundant to write `node:` for `getBuiltin`, and I gather that that’s what troubles Joyee. But it’s quite a mouthful to document why `node:` is forbidden for `getBuiltin` and yet sometimes required for `isBuiltin`: 

> For `getBuiltin`, the `node:` prefix is unnecessary and is therefore forbidden. However for `isBuiltin`, the `node:` prefix is required for builtin modules that are only accessible with the prefix, such as `node:test`, to disambiguate from user modules. Therefore, `isBuiltin('node:test')` is true, but `getBuiltin('node:test')` throws; and `isBuiltin('test')` returns false while `getBuiltin('test')` returns the builtin.

This makes some sort of sense, kinda, but overall the end result feels like a bug, doesn’t it? Maybe if `getBuiltin` existed in isolation, it would feel natural to never need the prefix; but since it’s _not_ in isolation, we need to consider its counterparts. It’s much simpler to explain that there’s one set of strings of builtin IDs, and it’s the same set used by all three of `isBuiltin`, `getBuiltin` and `builtinModules`. In other words, this test should pass:

```js
const { ok } = require('assert');
const { builtinModules, isBuiltin, getBuiltin } = require('module');

const builtinIds = ['fs', 'node:fs', 'node:test'] // Plus everything where `isBuiltin(id) === true`
for (const id of builtinIds) {
  ok(builtinModules.includes(id));
  ok(isBuiltin(id));
  ok(getBuiltin(id));
  ok(process.getBuiltin(id));
}
```

There’s already an [open issue](https://github.com/nodejs/node/issues/42785) for `builtinModules` missing `node:test`, and from what I gather reading that issue there’s no debate about it; someone just needs to open a PR to add it (and also `node:`-prefixed versions of the older builtins like `fs`).

Reply:
I updated it to accept the same kind of ID accepted by `module.isBuiltin()`. Though still looks weird to me, I don't think I really care about it that much as long as it's other people making that decision..